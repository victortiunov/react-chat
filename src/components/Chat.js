import React from 'react';
import { withStyles } from 'material-ui/styles'
import ChatMessageList from './ChatMessageList';
import MessageInput from './MessageInput';

const styles = theme => ({
	chatLayout: {
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		paddingTop: '64px',
		height: '100%',
		width: '100%',
		overflow: 'hidden',
	}
});

class Chat extends React.Component {
	render() {
		const { classes, user, activeChat, sendMessage, joinChat } = this.props;

		return (
			<main className={classes.chatLayout}>
				{ activeChat ? (
					<React.Fragment>
						<ChatMessageList user={user} messages={activeChat.messages} />
						<MessageInput sendMessage={(message) => sendMessage(activeChat._id, message)} user={user} onJoinClick={() => joinChat(activeChat._id)} />
					</React.Fragment>
				) : (
					<div></div>
				)}
			</main>
		);
	}
}

export default withStyles(styles)(Chat);
