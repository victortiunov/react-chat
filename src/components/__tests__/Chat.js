/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from '../Chat';

jest.mock('../ChatMessageList', () => () => 'ChatMessageList');
jest.mock('../MessageInput', () => () => 'MessageInput');

const mockProps = {
  user: {
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
  activeChat: {
    _id: '',
  },
  joinChat: jest.fn(),
  sendMessage: jest.fn(),
  isConnected: true,
};

describe('<Chat />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly ', () => {
    const tree = renderer.create(<Chat {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
