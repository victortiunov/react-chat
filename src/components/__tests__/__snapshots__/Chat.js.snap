// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders correctly  1`] = `
<main
  className="Chat-chatLayout-3"
>
  ChatMessageList
  MessageInput
</main>
`;
